Updated windows window/frontend to fix issue with html select positioning

MikeSchaap opened the pull request:
Fixes #1081

Need to wait for the following PR (https://github.com/leaanthony/go-webview2/pull/5) to merge so `go-webview2` exposes the `NotifyParentWindowPositionChanged()` method.

Was not sure how to correctly expose the chromium object to the window object. Do you agree with the proposed fix? Doesn't feel really solid imo

Note: First time creating a public Pull Request, so if I can improve certain parts, please let me know :)

leaanthony commented:
I would suggest having the notify function in the Window struct rather than options, as that's user options. In frontend Run, you can set the notify callback directly on the window struct after setting up chromium

MikeSchaap commented:
I moved the `notifyParentWindowPositionChanged` method to the Window struct and set the value in the `Run` method of `Frontend`. I was getting a nil pointer on the `WndProc` because it was called before the `notifyParentWindowPositionChanged` method was set. I moved the `winc.RegMsgHandler(w)` method call from the `NewWindow` method to the `Window.Run` method to make sure `notifyParentWindowPositionChanged` method was set before it was being used. Do you have any idea if this might interfere with anything else? From my (short) testing, everything seemed to still work.

MikeSchaap commented:
Also, I guess we still have to wait for a new tag on the `go-webview2` repo before we can update the version in `go.mod` right?

leaanthony commented on v2/internal/frontend/desktop/windows/window.go:
```diff
@@ -86,5 +86,16 @@ func NewWindow(parent winc.Controller, appoptions *options.App) *Window {
 }
 
 func (w *Window) Run() int {
+	winc.RegMsgHandler(w)
```
Could we put this back to where it was and deal with the nil pointer scenario?

leaanthony commented on v2/internal/frontend/desktop/windows/window.go:
```diff
@@ -86,5 +86,16 @@ func NewWindow(parent winc.Controller, appoptions *options.App) *Window {
+	case w32.WM_NCLBUTTONDOWN:
+		w32.SetFocus(w.Handle())
+	case w32.WM_MOVE, w32.WM_MOVING:
+		w.notifyParentWindowPositionChanged()
```
```suggestion
		if w.notifyParentWindowPositionChanged != nil {
			w.notifyParentWindowPositionChanged()
		}
```

leaanthony reviewed (changes requested):
I think this is almost ready to go 👍

leaanthony commented:
Also need to resolve conflicts

MikeSchaap commented:
Oh no, i'm pretty sure i messed up... I shouldn't work on sunday i guess.. Trying to figure out how to 'restore' this PR

MikeSchaap commented:
Looks like I managed to revert the messup. Also moved the `winc.RegMsgHandler(result)` back to it's original position and resolved the conflicts. I believe we should be good to go now

leaanthony commented:
> Also, I guess we still have to wait for a new tag on the `go-webview2` repo before we can update the version in `go.mod` right?

Nah, just update the dependency to use latest commit. I usually specify this like `go get .....@ab2c3b2`

stffabi commented:
> > Also, I guess we still have to wait for a new tag on the `go-webview2` repo before we can update the version in `go.mod` right?
> 
> Nah, just update the dependency to use latest commit. I usually specify this like `go get .....@ab2c3b2` Do you need to update go.mod + go.sum in this PR?

PR #1105 already bumped the `go-webview2` dependency and that PR has been merged. Therefore we could merge this PR without needing to bump it. 
Should we squash the commits before merging?

leaanthony commented:
LGTM!

MikeSchaap commented:
Cheers 🍾

leaanthony commented:
Congrats on your first PR!
